File: mog-rs/factory/postgres-tables-to-go/tests/expected.go
type Actor struct {
	ActorId int32 `db:"actor_id"`
	FirstName string `db:"first_name"`
	LastName string `db:"last_name"`
	LastUpdate time.Time `db:"last_update"`
}

type Staff struct {
	StaffId int32 `db:"staff_id"`
	FirstName string `db:"first_name"`
	LastName string `db:"last_name"`
	AddressId int32 `db:"address_id"`
	Email *string `db:"email"`
	StoreId int32 `db:"store_id"`
	Active bool `db:"active"`
	Username string `db:"username"`
	Password *string `db:"password"`
	LastUpdate time.Time `db:"last_update"`
	Picture []byte `db:"picture"`
}

type Widget struct {
	WidgetId int64 `db:"widget_id"`
	SmallCount int16 `db:"small_count"`
	BigCount *int64 `db:"big_count"`
	UnitPrice string `db:"unit_price"` // TODO(mog): Postgres exact-precision column; mapped to a Go string (no native fixed-point) -- parse with a fixed-point library for arithmetic.
	Ratio *float32 `db:"ratio"`
	PreciseRatio *float64 `db:"precise_ratio"`
	Sku string `db:"sku"`
	Grade *string `db:"grade"`
	Description *string `db:"description"`
	IsActive bool `db:"is_active"`
	CreatedOn time.Time `db:"created_on"`
	OpenTime *time.Time `db:"open_time"`
	MadeAt time.Time `db:"made_at"`
	UpdatedAt time.Time `db:"updated_at"`
	ExtId string `db:"ext_id"`
	Payload json.RawMessage `db:"payload"`
	Meta json.RawMessage `db:"meta"`
	Thumbnail []byte `db:"thumbnail"`
	Rating *string `db:"rating"` // TODO(mog): Postgres enum; represented as string in Go (define a named type + constants if you need type safety).
	Fulltext *string `db:"fulltext"` // TODO(mog): Postgres full-search vector; stored as string, not searchable in Go.
	SpecialFeatures []string `db:"special_features"` // TODO(mog): Postgres array; mapped to a Go slice -- verify the element type.
}

File: mog-rs/factory/postgres-tables-to-typescript/tests/expected.ts
export interface Actor {
  actor_id: number;
  first_name: string;
  last_name: string;
  last_update: string;
}

export interface Staff {
  staff_id: number;
  first_name: string;
  last_name: string;
  address_id: number;
  email: string | null;
  store_id: number;
  active: boolean;
  username: string;
  password: string | null;
  last_update: string;
  picture: Uint8Array | null;
}

export interface Widget {
  widget_id: number;
  small_count: number;
  big_count: number | null;
  unit_price: number;
  ratio: number | null;
  precise_ratio: number | null;
  sku: string;
  grade: string | null;
  description: string | null;
  is_active: boolean;
  created_on: string;
  open_time: string | null;
  made_at: string;
  updated_at: string;
  ext_id: string;
  payload: unknown | null;
  meta: unknown | null;
  thumbnail: Uint8Array | null;
  rating: string | null; // TODO(mog): Postgres enum; consider a TypeScript union type instead of string.
  fulltext: string | null; // TODO(mog): Postgres full-search vector; stored as string, not searchable in TypeScript.
  special_features: string[] | null; // TODO(mog): Postgres array; verify element mapping and chosen TypeScript type.
}
